src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -7101,6 +7101,199 @@ .locals init (delegate*<System.Span<char>, System.Span<char>> V_0)
 ");
         }
 
+        [Fact, WorkItem(45447, "https://github.com/dotnet/roslyn/issues/45447")]
+        public void LocalFunction_ValidStatic()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+unsafe class FunctionPointer
+{
+    public static void Main()
+    {
+        delegate*<void> a = &local;
+        a();
+
+        static void local() => System.Console.Write(""local"");
```

Comment:
Consider adding a test where the contents of this method body exist within a lambda expression. Verifying the rewriting works in a nested environment.